internal/servertype/data_source_test.go

```diff
@@ -67,6 +73,28 @@ func TestAccDataSourceList(t *testing.T) {
 	all := &servertype.DDataList{}
 	all.SetRName("all")
 
+	// To make this test more resilient against changes to the server types list, we
+	// pre-calculate the indices of the server types we check against.
+
+	// Acceptance tests are usually skipped by terraform, we need to handle this ourselves here.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skipf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc)
+		return
+	}
+	// Make sure that a token is available
+	teste2e.PreCheck(t)
+	client, err := testsupport.CreateClient()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	serverTypes, err := client.ServerType.All(context.Background())
+	if err != nil {
+		t.Fatalf("failed to get list of server types: %v", err)
+	}
+
+	indexCPX11 := slices.IndexFunc(serverTypes, func(serverType *hcloud.ServerType) bool { return serverType.Name == "cpx11" })
+	indexCPX21 := slices.IndexFunc(serverTypes, func(serverType *hcloud.ServerType) bool { return serverType.Name == "cpx21" })
```

Comment:
I think this is overkill, if you really want to keep this, please extract it in a function so we don't make this test more complex that it needs to be.

Reply:
I think updating the code by the end of September is better.